app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,25 @@
+class SalesController < ApplicationController
+
+  def new
+    @sale = Sale.new
+    @vendor = Vendor.find(params[:vendor_id])
+    @product = Product.find(params[:product_id])
+  end
+
+  def create
+    id = params[:product_id]
+    product = Product.find(id)
+    product.sales.create(
```

Comment:
I'm curious what made you decide to switch from `Sale.create` to `product.sales.create`? Was the association with the product not being created correctly?